server.py

```diff
@@ -5,11 +5,575 @@
+      },
+      {
+        "id": 28,
+        "restaurant_id": 10,
+        "name": "Steve",
+        "createdAt": 1504095567183,
+        "updatedAt": 1504095567183,
+        "rating": 5,
+        "comments": "Head to this laid-back Long Island City restaurant for beef tongue tacos, chicken smothered in a heady mole sauce, and a monster crab tostada. New York's only Michelin-starred Mexican restaurant is an especially cool choice for lunch during the week or drinks after work. Eater critic Ryan Sutton awarded this restaurant two stars."
+      },
+      {
+        "id": 29,
+        "restaurant_id": 10,
+        "name": "Rob",
+        "createdAt": 1504095567183,
+        "updatedAt": 1504095567183,
+        "rating": 5,
+        "comments": "The hype was real. Please go. Get the ceviche. And the tres leches. You're welcome"
+      },
+      {
+        "id": 30,
+        "restaurant_id": 10,
+        "name": "Jason",
+        "createdAt": 1504095567183,
+        "updatedAt": 1504095567183,
+        "rating": 4,
+        "comments": "For a Michelin star restaurant, it's fairly priced and the food is fairly good. Started with a strawberry margarita which was good in flavor but not much alcohol. Had the chicken enchiladas with salsa verde and it was really good. Great balance in flavor and a good portion. Extra tasty with their hot sauces. My wife had the lamb but it was a bit too salty for our taste. Although, it was cooked very well and fell off the bone. The highlight of the night was the tres leches cake. Probably the best I've ever had to be honest. Not too sweet and very milky. Overall, one of my top 3 favorite Mexican in NY."
+      }]
+
+
+@app.route('/restaurants', methods=['GET'])
+def getAllRestaurants():
+    return jsonify(restaurants)
+
+@app.route('/reviews', methods=['GET'])
+def getAllReviews():
+    return jsonify(reviews)
+
+@app.route('/restaurants/favorites', methods=['GET'])   
+def getFavoritedRestaurants():
+  results = []
+  for restaurant in restaurants:
+    if(restaurant.get("is_favorite") == True):
+      results.append(restaurant)
+  return jsonify(results)    
+
+@app.route('/restaurants/<int:restaurant_id>', methods=['GET']) 
+def getRestaurantByID(restaurant_id):
+  for restaurant in restaurants:
+    if(restaurant.get("id") == restaurant_id):
+      return jsonify(restaurant)  
+
+@app.route('/reviews/restaurant/<int:restaurant_id>', methods=['GET'])
+def getAllReviewsForRestaurant(restaurant_id):
+  results = []
+  for review in reviews:
+    if(review.get("restaurant_id") == restaurant_id):
+      results.append(review)
+  return jsonify(results) 
+
+@app.route('/reviews/remove/<int:review_id>', methods=['GET'])
+def removeRestaurantReview(review_id):
+  review_id = request.json.get("id")
+  for review in reviews:
+    if(review.get("id") == review_id):
+      reviews.remove(review)
+      break
+  return jsonify(reviews)  
```

Comment:
It seems odd to return all reviews in this situation, is this really what the other API did?

Reply:
No, we discussed this before. The original server used:
Delete REST method 
http://localhost:1337/reviews/<review_id>

But Delete is not supported by app.route/Flask. And that same endpoint is being used for another Post method (updating a review)

The og code for delete has the review id being passed in through the url, which I wanted to keep. So I used a Get method and had it return the remaining reviews.

Right now the only data being extracted from requests are reviews or restaurants, to change to a Post method you would have to put the review id in a mostly empty json file. Is that would you would prefer?
http://localhost:1337/reviews/remove
Request.json.get("id")
With the json being {"id": 3}

Reply:
Ok, well a GET is not appropriate
Regardless, don't return the reviews here, maybe the correct behavior is
OK - in success
NOT FOUND - for a bad I'd
FAILED - for failed